### DISCUSSION TITLE
Topology tests: parameter estimation for each user-defined tree

### DISCUSSION THREAD

**Discussion by rbjleroy** [iqtree/iqtree2#234] [Q&A] [answered]
Hi,

I am currently trying to test different topologies given an alignment using IQTREE.
I used to input in the command line '-te tree.nw' with tree.nw the ML tree found with the alignment. However, I noticed that parameter estimation is only done once on this inferred tree, before using the parameters for all other topologies.

Is there a way to make IQTREE estimate parameters for each user-defined topology in the '-z' argument? Or should I instead run each topology separately (make one 'topology test' per user-defined topology)?

In advance, thank you very much for your help and time!

Best,
Romain

**Comment by bqminh** [comment, marked as answer]
Yes, you can do that, via `--estimate-model` option. By default, IQ-TREE only optimizes branch lengths for each tree in `-z` because the aim is to compare topologies. So model parameters are not that important, as along as they are reasonably estimated on a fixed tree topology to save time.
